sdks/python/pyproject.toml

```diff
@@ -21,7 +21,7 @@
 requires = [
     "setuptools",
     "wheel>=0.36.0",
-    "grpcio-tools==1.62.1",
```

Comment:
these are requirements for beam building environment; this will not have an impact on dependency conflict you are resolving.

Reply:
This affects the version of grpc compiler used to generate python stubs for grpc protos, and we cannot use the latest version due to incompatibilites.

```
RuntimeError: The grpc package installed is at version 1.65.5, but the generated code in org/apache/beam/model/pipeline/v1/metrics_pb2_grpc.py depends on grpcio>=1.66.2. Please upgrade your grpc module to grpcio>=1.66.2 or downgrade your generated code using grpcio-tools<=1.65.5.
```

I would keep it as is, we MIGHT have to bump this up due to protobuf upgrade but if we don't have to, I think we can keep as is.

We definitely cannot upgrade to grpcio>=1.66.2 while https://github.com/grpc/grpc/issues/37710 is unresolved

cc: @jrmccluskey

Reply:
Pinned this to  1.65.5. Seem working now.